### DISCUSSION TITLE
Managing Expiring Access Tokens [help]

### DISCUSSION THREAD

**Discussion by daranzolin** [ropensci/targets#1343] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I have a targets workflow that begins with an API access token. The token expires every two hours so I can do something like:
```
tar_age(access_token, get_access_token(), age = as.difftime(2, units = "hours"))
```
The problem, however, is that the access token is only required in the next few targets, several API calls. But now every two hours everything else downstream becomes outdated. 

How can I configure the targets so that the API calls are (1) still dependent on the access token; but (2) don't need to rerun every two hours?

**Comment by wlandau** [comment, marked as answer]
You might consider a different way to mange the access token than to put it in a target. It's a bit of an odd fit because an update to the access token does not necessarily mean the data behind the API has changed.
